### DISCUSSION TITLE
Discussing the future of Orca

### DISCUSSION THREAD

**Discussion by julie777** [OrcaSlicer/OrcaSlicer#1156] [General]
I have been waiting to bring this up. I have had a vision about the future of the 3D printing ecosystem for some time now and I have been keeping notes based on using all of the various tools, different printers, different slicers, and so on.

When I first started using Orca Slicer, it was because of the built in calibration. It seemed to be just @SoftFever owning it, managing it, and doing all the development. I asked about the project goals and never got a response. 

Over time many printers were added. There was more and more activity and now the plan to change how profiles are managed makes it feel like the time is right to see if anyone feels like I do and wants to do something bigger. 

What I would like to do is design a better tool-chain and overall user experience from end to end. Do it in a way that directly supports all types of users, novice that just wants to print models they download up to the small business running a print farm. (I don't care about industrial printing. I believe that they have their own bubble that supports $50,000 printers and processes.)

I would like us to consider using Orca as a "brand" and a name for the community around this project.

I have been trying to think of a good name for a Reddit community to discuss the future, share ideas and start to build out a better system. By no means do I mean start from scratch. Instead incorporate the best from what exists and work with a plan to avoid wasting effort. 

Some examples of how the ecosystem could be improved.
  - a filament profile library - I see posts every day asking about the best settings for some filament
  - a machine profile library - this might not be directly portable to other slicers, but it would contain the machine information that everyone needs regardless of their slicer
  - a library of custom G code that optimizes certain things
  - a set of tools that work together, but is separate and can be used by itself. Some examples
      - the G code viewer that is part of Prusa Slicer. 
      - a packager and publisher that can automate the process of collecting the required files and uploading them to thangs, or Printables
      - a profile manager that can show dependencies and allow user to easily manage all of there custom profiles, diff and transfer, rename,create, delete.
  - an IDE that ties everything together. BBL is trying to do this with Bambu Studio, but I for one am not satisfied with their approach. I think we can do better.

I have spent time working on some profile management for myself because a change in firmware might cause a change in the system profile's gcode, which could cause my custom G code to have issues.
I have also been working to reverse engineer BBL gcode that they have not documented and add it to a master list that compares what various brands and firmware supports.
None of what I have been working on has been pushed to GitHub yet. 

If there is enough interest it this type of big picture master plan to help the community, I propose we create a Reddit community and start discussing the ideas.

What do you all think?

**Comment by julie777** [comment]
It just occurred to me that when you read this, the first question you will have is "why do I care what she thinks?". 

I am going to go update my GitHub profile so there will be more about me there, but I think I should give you a short summary of what I could bring to the Orca group.

I am a semi-retired software engineer. I have time now to work on OS projects that I feel passionate about. I contributed to the success of many commercial products and I know what it takes to create a great product, to maintain it and to make users happy. My last commercial product was SlingTV, which I contributed to at all levels from architecture to deployment automation.

**Comment by julie777** [comment]
This should be move to discussion

**Comment by SoftFever** [comment]
@julie777 
Thanks for sharing your thoughts.
I've considered the idea of an online profile repository, but I haven't had the time to work on it yet.
If you or any other developer is interested in working on it, that would be great, and we can continue the discussion in the [Dev](https://github.com/SoftFever/OrcaSlicer/discussions/categories/dev) section.

As an open-source project, there isn't and won't be an aggressive development plan for OrcaSlicer. I develop and maintain it in my spare time. All contributors so far are the same.
There isn't any so-called "closed" Orca group. 
Any developer can contribute to the project at their own comfortable pace and at any time. 
For small changes or features, pull requests (PRs) are sufficient.
If there are any significant proposed changes, it is necessary to have a discussion thread in [Dev](https://github.com/SoftFever/OrcaSlicer/discussions/categories/dev) beforehand.

If, in the future, OrcaSlicer attracts enough consistent contributors, I am open to adding collaborators or maintainers to the repository. Alternatively, we can consider moving it to an organization account for more convenient sharing of work.

Until then, I want to keep things simple.

PS. Lastly, I want to emphasize that OrcaSlicer will stay true to its principles(open source community project)

**Comment by julie777** [comment]
@SoftFever 
Sadly, I failed to convey my thoughts clearly. I agree with your thoughts about the project, as you stated in your previous comment.

@fr3ak2402 might also be interested in reading this.

My reason for a GitHub group was not to make things more formal or create a closed group. In fact, just the opposite. I was hoping to attract more contributors. You actually, support what I was trying to get across when you mention the profile repository.

I am going to try again to make myself understood. The purpose of a GitHub group is to allow multiple repositories related to the project to share a home in one place. Currently that is your home on GitHub which hosts the Orca Slicer repo along with any other repos that you create.

My reason for suggesting a group as the home instead of under your user was to allow more repos to be created under the Orca name, such as a profile repository. 

If one looks at Prusa Slicer, it is actually 2 separate applications: the slicer and the G code simulator. Maintaining those in separate repos would make maintenance easier and it could be done by different contributors. The builds are definitely simpler when smaller and more focused. 

BBL, in creating Bambu Studio, was thinking bigger than just a slicer, but they barely started most of what the "studio" part implies. They talk about projects, but any support currently is just a token/placeholder. They have very poor support for multiple printers, no job scheduler, a menu item for printing all plates, but printing all plates is not supported. Their problem is mostly having a large vision, but being focused on making sure the novice users can just print stuff, and needed to get the printer to market as quickly as possible.

In contrast, by using separate repos, and allowing multiple applications to be worked on independently, Orca could have a full features job management application, that could be integrated into the main IDE when ready, replacing the device page, and provide job queuing, prioritized printing, printer selection based on capability. It could be something that works for the user with one printer, or for someone with a printer farm. Unlike the device page, it would not be so specific to BBL printers, but much more useful to all users. This will be much harder to develop if it must be done as part of the slicer application and incrementally released in bits and pieces with each slicer release.

I have been thinking about an application for managing profiles, managing dependencies, allowing creating, modifying, deleting, diffing, etc. The addition of a new printer kind could be done by any user, provided no slicer work was necessary. Again, separate application under the Orca name, and eventually integrated into the main IDE if desired. This one is easy as a separate application given the current Studio/Orca opens most of the profile management in separate windows. Filaments are always in separate windows, as are machines and diffing.

Printer management is also a separate application that could be created. It could assume at least part of the role of profile management for the actual printer part. This would be the preferred place for a user to add a new printer definition. Even better is that the user could manage the printers they have, regardless of the number and model of them. When changing a nozzle to a different size on a given printer they could record it here. That way if that model of printer was selected prior to slicing and that was the only instance of that model existing, then the printer configuration for that printer is limited to variations with only that nozzle size. (I could write multiple pages about just this topic.)

I am sure you can see how there are many discrete functions of a full and complete 3D printing environment that could exist under the Orca name. I have not even mentioned project management, auto uploading to online repos, etc.

@SoftFever 
I think Orca is a cool name. I am a nerd from way back. I want to somehow bring more developers and enthusiasts together to create a **great** integrated environment for 3D printing that works for novice users, enthusiasts like us, and even small businesses running print farms. I am looking for your support to have that solution ultimately be a collection of Orca branded open source apps that all work together in Orca Studio, Orca IDE, or whatever the name for the full environment turns out to be.

If you choose to not be part of that kind of effort, that is okay. My original attempt to find out what your plans are for Orca was to see if you wanted to go this direction. I have great admiration for what you have already accomplished. It seems that Orca Slicer is at an inflection point where it is gaining a large following and no longer "just" Bambu Studio with calibration added. 

I know this is long. I could not find a way to say it more concisely. I think now that BBL has raised the bar for hardware, and many other companies are rushing to do the same, that this is the point where the entire 3D printing ecosystem needs to also have a big improvement in efficiency. If the right ideas are put out there and more of the enthusiast developers come together that would be a group that could do great things.

In contrast, there are many slicers out there, most of them open source. This means there are many groups of developers all working to make their project better and spending time re-implementing what another slicer has done. Resources are spread thin. Users have to choose among many possible tools and none of them has everything in one place. If I compare the 3D printing open source software space to the typical business world, there have been many startup with great ideas. Now the industry is starting to mature and in the business world that would mean consolidation. Companies would merge. Instead of working separately some of the brightest minds would start working together.

Okay, I have definitely gone too far. 
Just let me know if I should keep my focus on Orca and start work on one of the tools/apps that I mentioned, or whether I just "shove-off". 

**Regardless, thank you for starting something better than what existed by adding calibration into the slicer.**

**Reply by imagesurgery** [reply]
As a startup owner, coming from a wildly-different-but-developer-adjacent industry with highly managed software pipelines, and seeing the positive effect on not just productivity, but the entire indescribable 'flow' of not having to micromanage the details of a workflow _every single time_ you do the thing…

I absolutely love both the specific suggestion of modularising not just the existing OrcaSlicer, but the generalised need in the entire 3D printing workflow, hardware, history, management, calibration, extensible environment that this suggestion describes the current void.

And having seen top-tier software from great companies get repeatedly bought by some unstoppable force like Autodesk where it becomes a franken-monster version of its former greatness at increased cost yet inescapable necessity, the dedication in this space to the foundations of the entire activity of 3D printing, I seriously applaud your efforts.

Thank you!

**Comment by Pepe-Polymaker** [comment]
@julie777 Hi Julie. I just saw this post. What you described two years ago is something that we also have been thinking about. Are you still interested in any of those?

**Reply by dewi-ny-je** [reply]
I suggested in the past Vector3D to collect (now that he has an online tool for shrinkage) the shrinkage values for various printers and filaments, so that new users could input a reasonable value by selecting the printer and filament, without to even have to print the calibration piece.

Filament profiles could be similar. If there is a database (printer model and nozzle in use are important) it would be easier to pick starting settings.
